UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Helpers/FileSystemHelper.cs

```diff
@@ -119,6 +122,28 @@ public void CreateXmlFile(byte[] fileContent, string targetPath)
             _fileUtility.CreateXmlFile(fileContent, targetPath);
         }
 
+        public Task CreateXmlFromObject<T>(T obj, string filePath, string fileName)
```

Comment:
Suggestion: this method can be made generic. just pass the filename without extension.